dataland-batch-manager/src/main/kotlin/org/dataland/datalandbatchmanager/service/CompanyUploader.kt

```diff
@@ -15,34 +18,53 @@ import java.net.SocketTimeoutException
 @Service
 class CompanyUploader(
     @Autowired private val companyDataControllerApi: CompanyDataControllerApi,
+    @Autowired private val objectMapper: ObjectMapper,
 ) {
     companion object {
         const val MAX_RETRIES = 3
         const val UNAUTHORIZED_CODE = 401
-        const val FORBIDDEN_CODE = 403
     }
 
     private val logger = LoggerFactory.getLogger(javaClass)
 
-    /**
-     * Uploads a single Company to the dataland backend performing at most MAX_RETRIES retries.
-     * This function absorbs errors and logs them.
-     */
-    fun uploadSingleCompany(
-        companyInformation: CompanyInformation,
-    ) {
+    @Suppress("ReturnCount")
+    private fun checkForDuplicateIdentifierAndGetConflictingCompanyId(exception: ClientException): String? {
+        if (exception.statusCode != HttpStatus.BAD_REQUEST.value()) {
+            return null
+        }
+
+        val exceptionResponse = exception.response!! as ClientError<*>
+        val exceptionBodyString = exceptionResponse.body.toString()
+
+        val errorResponseBody = objectMapper.readTree(exceptionBodyString)
+        val firstError = errorResponseBody.get("errors").get(0)
+        if (firstError.get("errorType").textValue() != "duplicate-company-identifier") {
+            return null
+        }
+
+        val conflictingIdentifiers = firstError.get("metaInformation")
+        if (!conflictingIdentifiers.isArray || conflictingIdentifiers.size() != 1) {
```

Comment:
As we are only sending out one identifier at most one conflict should be able to occur. And in the error handling logic (before answering your comment) it is possible for an empty array to be returned here (in case of a race-condition in the backend). By returning null in that case, the upload would be re-tried and by then the race-condition will be resolved and the conflicting identifier will be returned.

Reply:
Aaah. We only store the LEI in the company information. Then this makes sense. So no other Identifieres thatn the LEI are retrievable via the GLEIF files?